### DISCUSSION TITLE
How to use Managed identity of durable function for AppInsight

### DISCUSSION THREAD

**Discussion by KreizIT** [Azure/azure-functions-durable-extension#2259] [Q&A]
Hello,

As AppInsight support Azure AD login (https://docs.microsoft.com/en-us/azure/azure-monitor/app/azure-ad-authentication?tabs=net), how to use this login method for durable orchestration / task ?

**Comment by davidmrdavid** [comment]
Hi @KreizIT, thanks for reaching out.
Can you please provide a bit more context on what you're trying to achieve? The question is a bit general at the moment.
In the meantime, have you reviewed our Managed Identity (preview) support documentation here: https://docs.microsoft.com/en-us/azure/azure-functions/durable/durable-functions-storage-providers#identity-based-connections-preview ?

Perhaps that already answers your question. Thanks!

**Reply by KreizIT** [reply]
Hi David,

Application Insight now support Azure Ad login for metrics. I just try to use this feature with durable function.
What I have done so far :
- Assign role "Monitoring Metrics Publisher"
- Disable local authentication in application insight

I strictly follow steps describe [here](https://docs.microsoft.com/en-us/azure/azure-monitor/app/azure-ad-authentication) except for this piece of code : 
```
services.Configure<TelemetryConfiguration>(config =>
{
       var credential = new DefaultAzureCredential();
       config.SetAzureTokenCredential(credential);
});
services.AddApplicationInsightsTelemetry(new ApplicationInsightsServiceOptions
{
    ConnectionString = "InstrumentationKey=00000000-0000-0000-0000-000000000000;IngestionEndpoint=https://xxxx.applicationinsights.azure.com/"
});

```

After local authentication is disabled in application insight, I do not have metrics from my app anymore.

So, is there a way for me to use the `new DefaultAzureCredential();` for TelemetryConfiguration with durable extension ?

**Reply by davidmrdavid** [reply]
I'll have to defer to @cgillum on this one. Chris, can you help?

**Comment by RohitRanjanMS** [comment]
Hi @KreizIT we don't support AD based authentication as of now. There is a plan to implement this for AI and for other services as well that supports AD based authentication. We are busy with other high priority items, but this is certainly something that we want to support. You can follow this issue to get an update on the progress.
